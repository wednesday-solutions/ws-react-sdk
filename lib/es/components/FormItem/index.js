import{_ as n,a as r}from"../_rollupPluginBabelHelpers-c2a46d82.js";import"antd/lib/form/style";import t from"antd/lib/form";import"../_rollup-plugin-inject-process-env-1259e43d.js";import e from"react";import o from"styled-components";import{c as a}from"../colors-93a064d8.js";import{m}from"../media-ee9cbc51.js";var i,p=o(t.Item)(i||(i=n(["\n  && {\n    .ant-form-item-explain,\n    .ant-form-item-extra {\n      text-align: left;\n    }\n    .ant-form-item {\n      margin: 0;\n    }\n    .ant-form-item-has-error .ant-input,\n    .ant-form-item-has-error .ant-input:hover,\n    .ant-form-item-has-error .ant-input,\n    .ant-form-item-has-error .ant-input:focus-within {\n      background-color: ",";\n      box-shadow: none;\n    }\n    .ant-form-item-has-error .ant-input-affix-wrapper,\n    .ant-form-item-has-error .ant-input-affix-wrapper:hover {\n      background-color: ",";\n      box-shadow: none;\n    }\n    &.ant-form-item-control-input {\n      min-height: 3.5rem;\n      ","\n      ","\n    width: 100%;\n    }\n  }\n"])),a.white,a.white,m.desktop.max("\n      min-height: 3rem;\n    "),m.largeMobile.max("\n      min-height: 2.5rem;\n    "));function f(n){return e.createElement(p,r({"data-testid":"form-item"},n))}f.propTypes={};export default f;
