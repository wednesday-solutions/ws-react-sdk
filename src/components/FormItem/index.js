/**
 *
 * FormItem
 *
 */

import React from 'react';
import { Form } from 'antd';
import styled from 'styled-components';
import colors from '@themes/colors';
import { media } from '@themes';

const FI = styled(Form.Item)`
  && {
    .ant-form-item-explain,
    .ant-form-item-extra {
      text-align: left;
    }
    .ant-form-item {
      margin: 0;
    }
    .ant-form-item-has-error .ant-input,
    .ant-form-item-has-error .ant-input:hover,
    .ant-form-item-has-error .ant-input,
    .ant-form-item-has-error .ant-input:focus-within {
      background-color: ${colors.white};
      box-shadow: none;
    }
    .ant-form-item-has-error .ant-input-affix-wrapper,
    .ant-form-item-has-error .ant-input-affix-wrapper:hover {
      background-color: ${colors.white};
      box-shadow: none;
    }
    &.ant-form-item-control-input {
      min-height: 3.5rem;
      ${media.desktop.max(`
      min-height: 3rem;
    `)}
      ${media.largeMobile.max(`
      min-height: 2.5rem;
    `)}
    width: 100%;
    }
  }
`;

function FormItem(props) {
  return <FI data-testid={'form-item'} {...props} />;
}

FormItem.propTypes = {};

export default FormItem;
