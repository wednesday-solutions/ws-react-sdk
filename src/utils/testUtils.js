import React from 'react';
import { IntlProvider } from 'react-intl';
import { render } from '@testing-library/react';
import { DEFAULT_LOCALE, translationMessages } from '@src/i18n';
import IntlGlobalProvider from '@components/IntlGlobalProvider';

export const renderWithIntl = children => {
  return render(
    <IntlProvider locale={DEFAULT_LOCALE} messages={translationMessages[DEFAULT_LOCALE]}>
      <IntlGlobalProvider>{children}</IntlGlobalProvider>
    </IntlProvider>
  );
};

export const getComponentStyles = (Component, props = {}) => {
  renderWithIntl(Component(props));
  const { styledComponentId } = Component(props).type;
  const componentRoots = document.getElementsByClassName(styledComponentId);
  // eslint-disable-next-line no-underscore-dangle
  return window.getComputedStyle(componentRoots[0])._values;
};

export const timeout = ms => new Promise(resolve => setTimeout(resolve, ms));
export const apiResponseGenerator = (ok, data) => ({
  ok,
  data
});

export default {
  timeout,
  renderWithIntl,
  getComponentStyles,
  apiResponseGenerator
};
