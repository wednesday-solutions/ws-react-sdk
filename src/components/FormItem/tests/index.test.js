/**
 *
 * Tests for FormItem
 *
 */

import React from 'react';
// import { fireEvent } from '@testing-library/dom'
import { renderWithIntl } from '@utils/testUtils';
import FormItem from '../index';

describe('<FormItem />', () => {
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<FormItem />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 FormItem component', () => {
    const { getAllByTestId } = renderWithIntl(<FormItem />);
    expect(getAllByTestId('form-item').length).toBe(1);
  });
});
