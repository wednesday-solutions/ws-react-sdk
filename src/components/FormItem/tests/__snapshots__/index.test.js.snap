// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormItem /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div
      class="ant-row ant-form-item sc-bdnxRM jntzBj"
      data-testid="form-item"
    >
      <div
        class="ant-col ant-form-item-control"
      >
        <div
          class="ant-form-item-control-input"
        >
          <div
            class="ant-form-item-control-input-content"
          />
        </div>
      </div>
    </div>
  </div>
</body>
`;
